components/script/script_runtime.rs

```diff
@@ -436,12 +445,29 @@ unsafe fn new_rt_and_cx_with_parent(parent: Option<ParentRuntime>) -> Runtime {
     DisableIncrementalGC(cx);
 
     unsafe extern "C" fn dispatch_to_event_loop(
-        _closure: *mut c_void,
-        _dispatchable: *mut Dispatchable,
+        closure: *mut c_void,
+        dispatchable: *mut JSRunnable,
     ) -> bool {
-        false
+        let networking_task_src: &NetworkingTaskSource = &*(closure as *mut NetworkingTaskSource);
+        let runnable = Runnable(dispatchable);
+        let task = task!(dispatch_to_event_loop_message: move || {
+            runnable.run(RustRuntime::get(), Dispatchable_MaybeShuttingDown::NotShuttingDown);
+        });
+
+        match networking_task_src.queue_unconditionally(task) {
+            Ok(_q) => return true,
+            Err(_) => return false,
+        };
+    }
+
+    if let Some(source) = networking_task_source {
```

Comment:
Actually, since you have the `JSContext` available here, you could do 
```rust
let global = GlobalScope::from_context(*cx);
let networking_task_source = global.networking_task_source();
```

I even wonder if you couldn't find a way to pass a `(task_source, canceller)` as the `*mut c_void`, so as to plug the queued task into a potential cancellation. 

The way to access it is by `global.task_canceller(TaskSourceName::Networking)`.

Reply:
Actually forget about potentially cancelling the task, if anything, it appears that at this point we should ascertain instead the queued task will run, in the light of https://github.com/servo/mozjs/blob/b2f83932fe9d361face14efd03f2465b9262e687/mozjs/js/public/Promise.h#L531

Reply:
There is no guarantee that there is a GlobalScope available when this code runs.